.github/workflows/lint.yml

```diff
@@ -0,0 +1,97 @@
+name: lint
```

Comment:
@moul What I did here was move misc -> lint and make misc.yml for the `misc` directory. And to me, it makes sense to call this one `lint` instead of `codegen`: aside from the `lint:` step, it has steps for `fmt`, `modtidy` and `generated`, all of which operate roughly the same (execute a command and check if the resulting source tree is different from the git tree). And I think it makes sense to still call them linting: they point out to how the code fixed should be with minimal and often automated effort on part of the user.

Reply:
Your `lint.yml` runs on every commit, including minor README changes or gno contract edits in the examples/ folder. Due to golangci-lint's intensity, I recommend maintaining it within each component's lint step: `tm2>lint`, `gnovm>lint`, `example>lint`.

We could introduce your new jobs to check codegen results across the entire repo. If so, naming it `codegen` seems more fitting than `lint`.

Currently, our workflows focus on individual components for clear context (easily identifying if a change affects only 'tm2' or the whole system). This differs from a single `lint` workflow, which would require deeper inspection to pinpoint the affected component.

Reply:
Okay, I propose this:

- Change this PR's `lint.yml` to run on changes to `**/*.go`, `**/*.gno`, `**/go.mod`, `**/go.sum`.
  - We include gno files because the fmt job also performs fmt checks on gno files.
- The job `lint` remains; it is supposed to have 0 issues all the time when we merge a PR, and it executes quickly, so I don't think it's necessary to divide it into each component.
- `fmt` and `modtidy` also remain, as they are also deterministic and IMO better if checked anytime we make any changes to the above pattern.
- As for codegen, I'll put in a new workflow, running only for codegen changes inside of `stdlibs`, `genstd` and the generated `native.go`.

Sounds good?

Reply:
waiting for @moul

Reply:
LGTM